v4l2.go

```diff
@@ -311,6 +336,56 @@ func getFrameSize(fd uintptr, index uint32, code uint32) (frameSize FrameSize, e
 	return
 }
 
+func getFrameInterval(fd uintptr, index uint32, code uint32, width uint32, height uint32) (FrameRate, error) {
+	frmivalEnum := &v4l2_frmivalenum{
+		index:        index,
+		pixel_format: code,
+		width:        width,
+		height:       height,
+	}
+
+	var frameRate FrameRate
```

Comment:
instead of setting the var here, in error cases you can do `return FrameRate{}, err`. for the two switch cases you can initialize and return directly

Reply:
Sounds good - now initializing FrameRate in return statements.